tests/test_crypto/test_ledger_apis.py

```diff
@@ -48,56 +54,102 @@ def test_initialisation(self):
-    #
-    #     with mock.patch.object(ledger_apis, 'transfer',
-    #                            return_value= "97fcacaaf94b62318c4e4bbf53fd2608c15062f17a6d1bffee0ba7af9b710e35"):
-    #         tx_digest = ledger_apis.transfer(FETCHAI, fet_obj, fet_address, amount=10, tx_fee=10)
-    #         assert tx_digest is not None
-    #         with mock.patch.object(ledger_apis, 'is_tx_settled', return_value= True):
-    #             assert ledger_apis.is_tx_settled(identifier=FETCHAI, tx_digest=tx_digest, amount=10)
-    #         with mock.patch.object(ledger_apis, 'is_tx_settled', return_value= False):
-    #             assert not ledger_apis.is_tx_settled(identifier=FETCHAI, tx_digest=tx_digest, amount=10)
-    #     with mock.patch.object(ledger_apis, 'transfer',
-    #                            return_value="97fcacaaf94b62318c4e4bbf53fd2608c15062f17a6d1bffee0ba7af9b710e35"):
-    #         tx_digest = ledger_apis.transfer(ETHEREUM, eth_obj, eth_address, amount=10, tx_fee=200000)
-    #         assert tx_digest is not None
-    #         with mock.patch.object(ledger_apis, 'is_tx_settled', return_value= True):
-    #             assert ledger_apis.is_tx_settled(identifier=FETCHAI, tx_digest=tx_digest, amount=10)
-    #         with mock.patch.object(ledger_apis, 'is_tx_settled', return_value= False):
-    #             assert not ledger_apis.is_tx_settled(identifier=FETCHAI, tx_digest=tx_digest, amount=10)
-    # def test_try_to_instantiate_fetchai_ledger_api(self):
-    #     """Test the instantiation of the fetchai ledger api."""
-    #     _try_to_instantiate_fetchai_ledger_api(addr="127.0.0.1", port=80)
-
-    # def test__try_to_instantiate_ethereum_ledger_api(self):
-    #     """Test the instantiation of the ethereum ledger api."""
-    #     _try_to_instantiate_ethereum_ledger_api(addr="127.0.0.1", port=80)
+            LedgerApis({"UNKNOWN": unknown_config})
+
+    def test_token_balance(self):
+        """Test the token_balance for the different tokens."""
+        ledger_apis = LedgerApis({ETHEREUM: DEFAULT_ETHEREUM_CONFIG,
+                                  FETCHAI: DEFAULT_FETCHAI_CONFIG})
+
+        with mock.patch.object(ledger_apis, 'token_balance', return_value=10):
+            balance = ledger_apis.token_balance(FETCHAI, fet_address)
+            assert balance == 10
+            balance = ledger_apis.token_balance(ETHEREUM, eth_address)
+            assert balance == 10
+
+        balance = ledger_apis.token_balance(FETCHAI, eth_address)
+        assert balance == 0, "This must be 0 since the address is wrong"
+        balance = ledger_apis.token_balance(ETHEREUM, fet_address)
+        assert balance == 0, "This must be 0 since the address is wrong"
+
+        with pytest.raises(AssertionError):
+            balance = ledger_apis.token_balance("UNKNOWN", fet_address)
+            assert balance == 0, "Unknown identifier so it will return 0"
+
+    def test_transfer(self):
```

Comment:
could this test be split in `test_transfer_fetchai` and `test_transfer_ethereum`